x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/RestEsqlTestCase.java

```diff
@@ -440,6 +440,38 @@ public void testOutOfRangeComparisons() throws IOException {
         }
     }
 
+    public void testInternalRange() throws IOException {
+        final int NUM_SINGLE_VALUE_ROWS = 100;
+        bulkLoadTestData(NUM_SINGLE_VALUE_ROWS);
```

Comment:
Suggestion: the test only creates single values, and the predicates are all trivial (i.e. we expect the filter to not filter out anything). I think it'd be more interesting if there was a couple of multivalued elements as well, and the predicate e.g. was true for something like half of them?

`testOutOfRangeComparisons` additionally creates some documents with MVs and nulls, which we could steal here.

Reply:
MVs and nulls are added.